src/filesystem/impls/filer/ArchiveUtils.js

```diff
@@ -263,6 +269,7 @@ define(function (require, exports, module) {
     }
 
     exports.archive = archive;
+    exports.archiveWithPath = archiveWithPath;
```

Comment:
Let's just change how `archive` works, so it always takes a path, and fix every call site that is currently not passing one in.